impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelProcessor.java

```diff
@@ -114,9 +114,9 @@ public Model read(XmlReaderRequest request) throws IOException {
             }
             try {
                 return doRead(request);
-            } catch (IOException e) {
-                exceptions.forEach(e::addSuppressed);
-                throw e;
+            } catch (RuntimeException ex) {
```

Comment:
Per Effective Java don't catch a raw runtime exception, only the specific subclass you expect to be thrown